internal/exec/stages/files/units_test.go

```diff
@@ -74,3 +83,245 @@ func TestParseInstanceUnit(t *testing.T) {
 		}
 	}
 }
+
+func TestSystemdUnitPath(t *testing.T) {
+
+	var logg log.Logger = log.New(true)
+	var st stage
+
+	st.DestDir = "/"
+	st.Logger = &logg
+
+	tests := []struct {
+		in  types.Unit
+		out []string
+	}{
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("system")},
+			[]string{"etc/systemd/system"},
+		},
+		{
+			types.Unit{Name: "test.service"},
+			[]string{"etc/systemd/system"},
+		},
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("global")},
+			[]string{"etc/systemd/user"},
+		},
+	}
+
+	for i, test := range tests {
+		path := st.SystemdUnitPaths(test.in)
+		if path[len(path)-1] != test.out[len(test.out)-1] {
+			t.Errorf("#%d: bad error: want %v, got %v", i, test.out, path)
+		}
+	}
+}
+
+func TestSystemdDropinsPaths(t *testing.T) {
+
+	var logg log.Logger = log.New(true)
+	var st stage
+
+	st.DestDir = "/"
+	st.Logger = &logg
+
+	tests := []struct {
+		in  types.Unit
+		out []string
+	}{
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("system")},
+			[]string{"etc/systemd/system/test.service.d"},
+		},
+		{
+			types.Unit{Name: "test.service"},
+			[]string{"etc/systemd/system/test.service.d"},
+		},
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("global")},
+			[]string{"etc/systemd/user/test.service.d"},
+		},
+	}
+
+	for i, test := range tests {
+		path := st.SystemdDropinsPaths(test.in)
+		if path[len(path)-1] != test.out[len(test.out)-1] {
+			t.Errorf("#%d: bad error: want %v, got %v", i, test.out, path)
+		}
+	}
+}
+
+func TestSystemdPresetPath(t *testing.T) {
+
+	var logg log.Logger = log.New(true)
+	var st stage
+
+	st.DestDir = "/"
+	st.Logger = &logg
+
+	tests := []struct {
+		in  types.Unit
+		out string
+	}{
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("system")},
+			"etc/systemd/system-preset/20-ignition.preset",
+		},
+		{
+			types.Unit{Name: "test.service"},
+			"etc/systemd/system-preset/20-ignition.preset",
+		},
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("user")},
+			"etc/systemd/user-preset/21-ignition.preset",
+		},
+		{
+			types.Unit{Name: "test.service", Scope: cfgutil.StrToPtr("global")},
+			"etc/systemd/user-preset/20-ignition.preset",
+		},
+	}
+
+	for i, test := range tests {
+		path := st.SystemdPresetPath(util.GetUnitScope(test.in))
+		if path != test.out {
+			t.Errorf("#%d: bad error: want %v, got %v", i, test.out, path)
+		}
+	}
+}
+
+func TestCreateUnits(t *testing.T) {
```

Comment:
We generally don't have unit tests that require root or that modify the filesystem; we use blackbox tests for that.  This might be okay, but I think it's surprising to have it here.

Reply:
I did that because I work in a Golang container that can't (or at least I don know how) run blackbox test...
I was written in the same way as https://github.com/coreos/ignition/blob/0cbe33fee45d012515479a88f0fe94ef58d5102b/internal/exec/util/user_group_lookup_test.go#L59
On the other hand, It was use for debugging purpose :)

Reply:
Oh, I didn't know about that code.  That's a different case though; it's testing some of the C bindings, not the general behavior of some Ignition feature.

During development, of course feel free to write code however is convenient for you.  Let's get this cleaned up before merging, though.

Reply:
ok, will remove that before merging :)